code/modules/mob/living/carbon/alien/humanoid/humanoid.dm

```diff
@@ -35,7 +36,10 @@
 
 	return 0
 
-///mob/living/carbon/alien/humanoid/bullet_act(obj/item/projectile/Proj) taken care of in living
+/mob/living/carbon/alien/humanoid/CanContractDisease(datum/disease/D)
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_VIRUSIMMUNE))
+		return FALSE
```

Comment:
why not just return false? seems weird to add a trait that is never removed then check  for that trait.

why have an if condition if its always true.

Reply:
That .... is a very good question. Hold on a second